src/typescript/Scala.tmLanguage.ts

```diff
@@ -118,7 +118,7 @@ export const scalaTmLanguage: TmLanguage = {
           name: 'constant.language.scala'
         },
         {
-          match: '\\b((0(x|X)[0-9a-fA-F]*)|(([0-9]+\\.[0-9]+)|(\\.[0-9]+))((e|E)(\\+|-)?[0-9]+)?|[0-9]+)([LlFfDd]|UL|ul)?\\b',
+          match: '\\b((0(x|X)[0-9a-fA-F_]*)|(([0-9][0-9_]*\\.[0-9_]+)|(\\.[0-9]+))((e|E)(\\+|-)?[0-9][0-9_]*)?|[0-9][0-9_]*)([LlFfDd]|UL|ul)?\\b',
```

Comment:
Is `.123_456` supported by this? Looking at the regex, that seems to not be the case.

Reply:
I will add a test for it and fix it if needed.

Reply:
I changed the regex